From df3915f9683c6aa42c6998831a4ae2382cc75597 Mon Sep 17 00:00:00 2001
From: Xinliang Liu <xinliang.liu@linaro.org>
Date: Thu, 27 Apr 2017 10:37:53 +0800
Subject: [PATCH 1/3] Make web frontend from https to http asscess

Https need certicates which repuire dns name to generate.
If you have no dns name for the web frontend server just IP access,
please apply this patch.

Signed-off-by: Xinliang Liu <xinliang.liu@linaro.org>
---
 ansible/group_vars/all                           |  2 +-
 ansible/roles/web-frontend/templates/apache.conf | 12 ++++++------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/ansible/group_vars/all b/ansible/group_vars/all
index 003b37c..676dae9 100644
--- a/ansible/group_vars/all
+++ b/ansible/group_vars/all
@@ -93,7 +93,7 @@ heat_cfm_private_api: "http://{{heat_host}}:8000/v1/%(tenant_id)s"
 front_end_ip: "*"
 front_end_services:
   - name: horizon
-    port: 443
+    port: 80
     host: "{{horizon_host}}"
     remote_port: 80
     preserve_host: true
diff --git a/ansible/roles/web-frontend/templates/apache.conf b/ansible/roles/web-frontend/templates/apache.conf
index b39bb75..7396efd 100644
--- a/ansible/roles/web-frontend/templates/apache.conf
+++ b/ansible/roles/web-frontend/templates/apache.conf
@@ -2,16 +2,16 @@
 
 {% for service in front_end_services %}
 # {{service.name}}
-{% if service.port != 443 %} {# Apache already binds to 443 so its not needed #}
+{% if service.port != 80 %} {# Apache already binds to 443 80 so its not needed #}
 Listen {{service.port}}
 {% endif %}
 <VirtualHost {{front_end_ip}}:{{service.port}}>
 	ServerName {{inventory_hostname}}
 
-	SSLEngine On
-	SSLCertificateFile {{ssl_cert}}
-	SSLCertificateKeyFile {{ssl_key}}
-	SSLCertificateChainFile {{ssl_ca_cert}}
+	# SSLEngine On
+	# SSLCertificateFile {{ssl_cert}}
+	# SSLCertificateKeyFile {{ssl_key}}
+	# SSLCertificateChainFile {{ssl_ca_cert}}
 
 	CustomLog /var/log/apache2/devcloud-{{service.name}}-access.log combined
 	ErrorLog /var/log/apache2/devcloud-{{service.name}}-error.log
@@ -21,7 +21,7 @@ Listen {{service.port}}
 
 	# force an app like horizon/django to send urls back with HTTPS.
 	# otherwise wsgi end of this proxy thinks its handling HTTP links
-	RequestHeader set X-Forwarded-Proto 'https' env=HTTPS
+	# RequestHeader set X-Forwarded-Proto 'https' env=HTTPS
 {% if service.preserve_host is defined %}
 	ProxyPreserveHost On
 {% endif %}
-- 
2.1.4

